### PR TITLE
Migrate Away from Docker Hub

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Open source is the future! And AWS too I guess...

Use Colima (`brew install colima`) to run locally. See Colima issue 44 for instructions on how to use `docker buildx` locally.

### CODE REVIEW

File: .github/workflows/maven-ci.yml [hunk 1]
```diff
@@ -104,16 +104,32 @@ jobs:
...
           context: .
           platforms: linux/amd64,linux/arm64
           push: true
-          # Temporary! Update to real account
-          tags: milesgrantibigroup/otp-middleware-test:${{ github.sha }}, milesgrantibigroup/otp-middleware-test:latest
\ No newline at end of file
+          add_git_labels: true
+          tag_with_ref: true
+          tags: public.ecr.aws/s2a5w2n9/otp-middleware:${{ github.sha }}
+      - name: Build and push with different tag
```
Comment (binh-dam-ibigroup) [inline .github/workflows/maven-ci.yml:125]: What was the purpose to run this other build?

Comment (miles-grant-ibigroup) [reply .github/workflows/maven-ci.yml:125]: Docker hub allows you to create multiple tags at once (one for the commit hash, one to overwrite `latest`). Amazon however requires multiple requests

File: .github/workflows/maven-ci.yml [hunk 2]
```diff
@@ -104,16 +104,32 @@ jobs:
...
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          add_git_labels: true
+          tag_with_ref: true
```
Comment (binh-dam-ibigroup) [inline .github/workflows/maven-ci.yml]: These two parameters result in [warnings in the docker build step](https://github.com/ibi-group/otp-middleware/runs/7676427873?check_suite_focus=true#step:18:1). Are they needed?

File: .github/workflows/maven-ci.yml [hunk 3]
```diff
@@ -104,16 +104,32 @@ jobs:
...
         uses: docker/build-push-action@v2
         with:
           context: .
           platforms: linux/amd64,linux/arm64
           push: true
-          # Temporary! Update to real account
-          tags: milesgrantibigroup/otp-middleware-test:${{ github.sha }}, milesgrantibigroup/otp-middleware-test:latest
\ No newline at end of file
+          add_git_labels: true
+          tag_with_ref: true
```
Comment (binh-dam-ibigroup) [inline .github/workflows/maven-ci.yml]: These two parameters result in [warnings in the docker build step](https://github.com/ibi-group/otp-middleware/runs/7676427873?check_suite_focus=true#step:17:1). Are they needed?

**Review by binh-dam-ibigroup — APPROVED** [review]
Seems fine, although check or add a comment in `maven-ci.yml` regarding `add_git_labels` and `tag_with_ref` if they are needed.

File: .github/workflows/maven-ci.yml (hunk 2 above)
Comment (miles-grant-ibigroup) [reply .github/workflows/maven-ci.yml]: Nope! Good catch thanks

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I don't really know how to test this, but it looks fine to me.

Edit: I'm going to come back to this when I have better internet

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Colima works. Great work.

**Pull request merged**